dvc/repo/experiments/utils.py

```diff
@@ -342,3 +343,60 @@ def to_studio_params(dvc_params):
             result[file_name] = file_data.get("data", {})
 
     return result
+
+
+def _describe(
+    scm: "Git",
+    revs: Iterable[str],
+    logger,
+    refs: Optional[Iterable[str]] = None,
+    remove_ref_string=True,
```

Comment:
This function is used for `dvc exp list` and `dvc exp show`. For some reason `dvc exp show` uses the shortened path without the `ref/...` and `dvc exp list` uses the full path. A couple of tests break unless the output is formatted differently for each of them.

I haven't looked too much into it but it seems that simply rewriting the simple test for `dvc exp list` output is not enough, there are other things affected if we just keep the original `_describe`. Might be worth refactoring as the regex probably makes things slower (judging by the GH tests).

Reply:
I think what we really want is to make `experiments.ls()` only return git SHAs (so we would not call `describe()` at all in `dvc.repo.experiments.ls()`. At the UI/command level we can use `describe()` to get the names without the prefix when we actually want them.

So in `dvc.repo.experiments.ls` we can drop all of the git describe `name` handling entirely, so the returned result:
https://github.com/iterative/dvc/blob/43b371c43a090a5c05344f93538c7114a150319b/dvc/repo/experiments/ls.py#L51
now always only uses the baseline git SHA as the dict keys like:
```python
results[baseline].append(...)
```

The human-readable name handling would then be done strictly at the UI level, in `dvc.commands.experiments.ls`

Basically, we would replace this section:
https://github.com/iterative/dvc/blob/43b371c43a090a5c05344f93538c7114a150319b/dvc/commands/experiments/ls.py#L26-L35

with something like
```python
if name_only or sha_only:
    names = {}
else:
    names = describe(baseline for baseline in exps)
for baseline in exps:
    if not (name_only or sha_only):
        name = names.get(baseline) or baseline[:7]
        ui.write(f"{name}:")
    ...
```